### ISSUE TITLE
Allow splitting lists (creating two subsequent lists)

### ISSUE THREAD

**Issue by Reinmar** [ckeditor/ckeditor5-list#99]
## Current behaviour

Due to how the model representation of lists works, there's no way to create two subsequent lists of the same type.

In the model, we store only list items:

```html
<paragraph>foo</paragraph>
<listItem type=bulleted indent=0>fooa</listItem>
<listItem type=bulleted indent=1>foob</listItem>
<listItem type=bulleted indent=1>fooc</listItem>
<listItem type=bulleted indent=0>food</listItem>
<paragraph>foo</paragraph>
```

This means that in whichever way you'll modify the model (e.g. insert paragraph in the middle and then remove it or merge it into an existing list item), there will always be just one list.

Two subsequent lists are only possible to create if they have different `type`.

This normalised model structure makes it trivial to work with lists in the model. Enter, backspace, any insertion features (e.g. image upload) just insert whatever they want and wherever they want and the lists model to view conversion takes care of all the necessary changes. 

If you don't understand what "all necessary changes" mean, imagine how the DOM structure for the above mentioned list looks:

```html
<p>foo</p>
<ul>
	<li>fooa
		<ul>
			<li>foob</li>
			<li>fooc</li>
		</ul>
	</li>
	<li>food</li>
</ul>
<p>foo</p>
```

Now, imagine what happens when one indents the second list item ("foob"). Or when "fooc" is turned into a paragraph. And what needs to happen when this paragraph is deleted or turned back to a list item. Yes. It's awfully tricky.

Anyway, the side effect of such an implementation is that two subsequent lists are impossible to create now. This, in 99% cases is exactly what you expect. I can't, actually, think of a case when you'd like to have two subsequent lists one after another. 

Therefore, we're lucky that in CKEditor 5 we get this merging behaviour for "free". In CKEditor 4 lists are sometimes merged and sometimes not – it depends on how you created them, how you removed the content between them, etc. It's a pretty messy situation.

## Splitting lists

However, as we expected, the community brought this up that lists are always merged. For now, we only know about https://github.com/mozilla/notes/issues/641 but I'm quite certain that this will repeat.

For me, the most important questions are:

* Why you'd like to have two subsequent lists?
* How you'd expect to control this? When list items should be merged and when not?

As for the implementation details, a solution for that is introducing list ids to list items. Or any attribute which will mark a beginning of a list.

**Comment by Reinmar** [comment]
> As for the implementation details, a solution for that is introducing list ids to list items. Or any attribute which will mark a beginning of a list.

Actually, since it's a problem also from UX perspective to let the users enforce that two lists are not merged, perhaps it could be an element which we insert between two lists and which acts like a separator. It might be invisible in the data, but visible in the editor. And it'd need to be inserted consciously and intentionally.

**Comment by scofalik** [comment]
tl;dr

I am not sure how UX for this should work. I can imagine it like this:

1. Create a new list item of the same type after a list.
2. Enter in last, non-indented item could break lists before it is removed.
3. Removing empty list item between two list items of the same indent could lead to a state when lists are not merged. Would work well with <kbd>Enter</kbd>.

Of course merging will work naturally with <kbd>Backspace</kbd> and <kbd>Delete</kbd>.

**Comment by f1ames** [comment]
I stumbled upon this issue when working on Paste from Word (support for lists https://github.com/ckeditor/ckeditor5-paste-from-word/issues/5). While creating two subsequent lists in the editor may seem like an edge case, it may happen that such content is pasted from other sources (e.g. Word or [this codepen](https://codepen.io/f1ames/pen/OwqQOd?editors=1000#0)). Two subsequent lists are combined into on when pasted which seems like a buggy behaviour.

**Comment by Reinmar** [comment]
This is an incompatibility between the models (MS Word vs CKE5) and it's fine for now. The most important thing is that the content pasted into CKE5 gets adjusted to its model.
